elephant/conversion.py

```diff
@@ -152,3 +155,704 @@ def binarize(spiketrain, sampling_rate=None, t_start=None, t_stop=None,
+        if not all([type(elem) == neo.core.SpikeTrain for elem in spiketrains]):
+            raise TypeError(
+                "All elements of the input list must be neo.core.SpikeTrain "
+                "objects ")
+        # Link to input
+        self.lst_input = spiketrains
+        # Set given parameter
+        self.t_start = t_start
+        self.t_stop = t_stop
+        self.num_bins = num_bins
+        self.binsize = binsize
+        self.matrix_columns = num_bins
+        self.matrix_rows = len(spiketrains)
+        # Boolean, indicates if matrix shall stored
+        self.store_mat_u = store_mat
+        # Empty matrix for storage, unclipped matrix
+        self.mat_u = None
+        # Check all parameter, set also missing values
+        self.__calc_start_stop(spiketrains)
+        self.__check_init_params(binsize, num_bins, self.t_start, self.t_stop)
+        self.__check_consistency(spiketrains, self.binsize, self.num_bins,
+                                 self.t_start, self.t_stop)
+        # Variables to store the sparse matrices
+        self._sparse_mat_u = None
+        self._sparse_mat_c = None
+        # Now create unclipped version of sparse matrix
+        self.__convert_to_binned(spiketrains)
+
+    # =========================================================================
+    # There are four cases the given parameters must fulfill
+    # Each parameter must be a combination of following order or it will raise
+    # a value error:
+    # t_start, num_bins, binsize
+    # t_start, num_bins, t_stop
+    # t_start, bin_size, t_stop
+    # t_stop, num_bins, binsize
+    # ==========================================================================
+
+    def __check_init_params(self, binsize, num_bins, t_start, t_stop):
+        """
+        Checks given parameter.
+        Calculates also missing parameter.
+
+        Parameters
+        ----------
+        binsize : quantity.Quantity
+            Size of Bins
+        num_bins : int
+            Number of Bins
+        t_start: quantity.Quantity
+            Start time of the spike
+        t_stop: quantity.Quantity
+            Stop time of the spike
+
+        Raises
+        ------
+        ValueError :
+            If all parameters are `None`, a ValueError is raised.
+        TypeError:
+            If type of :attr:`num_bins` is not an Integer.
+        """
+        # Raise error if no argument is given
+        if binsize is None and t_start is None and t_stop is None \
+                and num_bins is None:
+            raise ValueError(
+                "No arguments given. Please enter at least three arguments")
+        # Check if num_bins is an integer (special case)
+        if num_bins is not None:
+            if type(num_bins) is not int:
+                raise TypeError("num_bins is not an integer!")
+        # Check if all parameters can be calculated, otherwise raise ValueError
+        if t_start is None:
```

Comment:
You don't actually raise a ValueError here.

The simplest approach would probably just be

```Python
if sum(x is None for x in [t_start, t_stop, num_bins, binsize]) > 1:
  raise ValueError('blah blah')
```